// Copyright 2011, Tencent Inc.
// Author: Yongsong Liu <ericliu@tencent.com>

option java_package="com.soso.poppy";

package poppy;

message RpcChannelOptions {
    // Type of Service http://en.wikipedia.org/wiki/Type_of_service
    optional int32 tos = 1;

    // keepalive time (idle connections will be closed after this time).
    // if there isnot payload in 79 seconds, disconnect the connection.
    // 79 is just a prime number.
    optional int32 keepalive_time = 2 [default = 79000]; // milliseconds

    // channel connection timeout, if a request can't get an healthy
    // connection after a connect_timeout milliseconds, it will fail
    // and return to the caller
    optional int32 connect_timeout = 3 [default = 10000]; // milliseconds

    // whether use channel cache to avoid create multiple RpcChannel
    // to same address, if true, multile RpcChannels to same address
    // will share only one RpcChannelImpl instance, otherwise,
    // we will create different RpcChannelImpl instance for each RpcChannel
    optional bool use_channel_cache = 4 [default = true];
}

